src/main/java/io/debezium/mcs/schema/ManagedConnectorsSchema.java

```diff
@@ -96,11 +103,34 @@ public void configure(Map<String, Object> config) {
         });
     }
 
+    private static Runtime.Version parseVersion(String version) {
+        Matcher m = VERSION_PATTERN.matcher(version);
+        if (m.matches()) {
+            return Runtime.Version.parse(version);
+        }
+        else if (m.lookingAt()) {
+            return Runtime.Version.parse(m.group());
+        }
+        throw new IllegalArgumentException("Invalid version string: \"" + version + "\"");
+    }
+
     @Override
     public String getSpec(org.eclipse.microprofile.openapi.models.media.Schema connectorSchema) {
-        ObjectMapper objectMapper = new ObjectMapper();
+        final ObjectMapper mapper = new ObjectMapper();
+
+        JsonNode fleetshardCatalog = mapper.valueToTree(new FleetshardCatalogEnvelope(connectorSchema));
+        final Map<String, Object> schemaExtensions = connectorSchema.getExtensions();
+        final Runtime.Version connectorVersion = parseVersion((String) schemaExtensions.get("version"));
+        final String overridesFilename = "overrides-" + schemaExtensions.get("connector-id") + "-" + connectorVersion.major() + "." + connectorVersion.minor() + ".json";
+        final InputStream overridesFileInputStream = getClass().getClassLoader().getResourceAsStream(overridesFilename);
+        if(null == overridesFileInputStream) {
+            throw new RuntimeException("Overrides file not found: " + overridesFilename);
+        }
+
+        var merger = mapper.readerForUpdating(fleetshardCatalog);
         try {
-            return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(new FleetshardCatalogEnvelope(connectorSchema));
+            merger.readTree(overridesFileInputStream);
```

Comment:
Updating the generated FleetshardCatalog with values from the override files. This way we can override multiple things in the entire ConnectorType descriptor.